### PR TITLE
Fix some small mistakes in stagnet3 pipeline

### PR DESCRIPTION
**Pull request by daniel1302**
### Changes

- Wrapped paths in slashes(`'`), which are required if the working directory in Jenkins contains space.
- Moved `sync local to remote` step, because we need to sync at very end of the pipeline when all steps are finished.

### Tests

Passed pipeline: https://jenkins.ops.vega.xyz/job/private/job/Deployments/job/Vegacapsule/job/Stagnet%203/16/console

Failed pipelines: All pipelines before 15:  https://jenkins.ops.vega.xyz/job/private/job/Deployments/job/Vegacapsule/job/Stagnet%203/

### CODE REVIEW

File: vars/capsulePipelineWrapper.groovy [hunk 1]
```diff
@@ -1,45 +1,45 @@
 def call() {
     writeConfigs = { envName ->
-        sh label: "generate templates: ${envName}", script: '''
+        sh label: "generate templates: ${envName}", script: """
             vegacapsule template genesis \
-                --home-path ${CONFIG_HOME} \
-                --path ${TEMPLATES_HOME}/genesis.tmpl.json ${FLAGS}
+                --home-path '""" + CONFIG_HOME + """' \
```
Comment (fkondej) [inline vars/capsulePipelineWrapper.groovy:5]: Can you tell me what's the advantage of this escaping over one of the below:
1. Simply using `’`
2. `${env.CONFIG_HOME}`

File: vars/capsulePipelineWrapper.groovy [hunk 2]
```diff
@@ -145,10 +145,11 @@ def call() {
                     stage('Sync remote state to local') {
                         steps {
                             sh """
-                                mkdir -p ${env.CONFIG_HOME}
-                                aws s3 sync ${env.S3_CONFIG_HOME}/ ${env.CONFIG_HOME}/
+                            set -x;
```
Comment (fkondej) [inline vars/capsulePipelineWrapper.groovy:148]: FYI you can define that you use bash (by default bash is not used).
Eg https://github.com/vegaprotocol/jenkins-shared-library/blob/main/vars/pipelineDevnetDeploy.groovy#L166

Comment (daniel1302) [reply vars/capsulePipelineWrapper.groovy:148]: Ahh sorry set -x is just testing. Let me remove it

File: vars/capsulePipelineWrapper.groovy [hunk 3]
```diff
@@ -229,11 +219,11 @@ def call() {
             stage('Restart Network') {
                 when {
                     expression {
-                        params.ACTION == 'RESTART' && params.UNSAFE_RESET_ALL
+                        params.ACTION == 'RESTART' && params.UNSAFE_RESET_ALL // TODO: This probably should be applied to START as well
```
Comment (fkondej) [inline vars/capsulePipelineWrapper.groovy:222]: about the comment: I would say that we should also allow a clean start, without `unsafe-reset-all`.

File: vars/capsulePipelineWrapper.groovy [hunk 4]
```diff
@@ -243,7 +233,17 @@ def call() {
                     }
                 }
                 steps {
-                    sh "vegacapsule network start --home-path ${env.CONFIG_HOME}"
+                    sh "vegacapsule network start --home-path '${env.CONFIG_HOME}'"
+                }
+            }
+            stage('Write to s3'){
```
Comment (fkondej) [inline vars/capsulePipelineWrapper.groovy:239]: Do you think you should also move the `'Write to git'` stage to the end?

**Review by fkondej — APPROVED** [review]
lgtm

File: vars/capsulePipelineWrapper.groovy (hunk 3 above)
Comment (daniel1302) [reply vars/capsulePipelineWrapper.groovy:222]: Yes yes currently we have:

- Start 
- Restart
- Restart + Unsafe reset all
- Stop :)

**Pull request merged**